web/src/features/charts/graphUtils.ts

```diff
@@ -193,19 +193,38 @@ export function getElectricityProductionValue({
   return generationTypeStorage === 0 ? 0 : -generationTypeStorage;
 }
 
+function analyzeChartData(chartData: AreaGraphElement[]) {
+  let estimatedCount = 0;
+  let tsaCount = 0;
+
+  for (const day of chartData) {
+    if (day.meta.estimationMethod === EstimationMethods.TSA) {
+      tsaCount++;
+    }
+    if (day.meta.estimationMethod || day.meta.estimatedPercentage === 100) {
+      estimatedCount++;
+    }
+  }
+
+  return {
+    allTimeSlicerAverageMethod: tsaCount === chartData.length,
+    allEstimated: estimatedCount === chartData.length,
+    hasEstimation: estimatedCount > 0,
+  };
+}
```

Comment:
Looks way better than all the redundat loops!

Reply:
That functional evangelist in me really struggled typing `let` 😆